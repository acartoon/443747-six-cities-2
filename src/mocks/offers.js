export const OFFERS = [
  {
    id: 154,
    title: `Beautiful & luxurious apartment at great location`,
    price: 120,
    isPremium: true,
    img: `img/apartment-01.jpg`,
    type: `Apartment`,
    coords: [52.3909553943508, 4.85309666406198],
  },
  {
    id: 114,
    title: `Wood and stone place`,
    price: 80,
    isPremium: false,
    img: `img/room.jpg`,
    type: `Private room`,
    coords: [52.369553943508, 4.85309666406198],
  },
  {
    id: 879,
    title: `Canal View Prinsengracht`,
    price: 132,
    isPremium: false,
    img: `img/apartment-02.jpg`,
    type: `Apartment`,
    coords: [52.3909553943508, 4.929309666406198],
  },
  {
    id: 123,
    title: `Nice, cozy, warm big bed apartment`,
    price: 180,
    isPremium: false,
    img: `img/apartment-03.jpg`,
    type: `Apartment`,
    coords: [52.3809553943508, 4.939309666406198],
  }
];
