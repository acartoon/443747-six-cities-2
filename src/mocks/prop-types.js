import PropTypes from "prop-types";

export default PropTypes.shape({
    id: PropTypes.number.isRequired,
    city: {
      name: PropTypes.string.isRequired,
      location: PropTypes.arrayOf(PropTypes.number).isRequired,
    }.isRequired,
    preview_image: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    is_premium: PropTypes.bool.isRequired,
    type: PropTypes.oneOf([`Apartment`, `Private room`]),
    price: PropTypes.number.isRequired,
    location: {
      latitude: PropTypes.number.isRequired,
      longitude:  PropTypes.number.isRequired,
      zoom: PropTypes.number.isRequired
    }.isRequired,
});