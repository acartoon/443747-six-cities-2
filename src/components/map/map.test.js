import React from "react";
import renderer from "react-test-renderer";
import Map from "./map";
import {OFFERS} from "../../mocks/offers";

it(`Map component render correct`, () => {
  const tree = renderer
      .create(
          <Map
            offers = {OFFERS}
          />
      )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
