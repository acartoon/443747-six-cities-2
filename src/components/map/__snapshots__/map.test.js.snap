// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceList component render correct 1`] = `
<section
  className="cities__map map"
>
  <div
    id="map"
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</section>
`;
